interface/wx/filedlg.h

```diff
@@ -112,12 +112,15 @@ const char wxFileSelectorDefaultWildcardStr[];
     display of the filter choice set the corresponding wxSystemOptions before calling
     the file open dialog:
     @code
-         wxSystemOptions::SetOption(wxOSX_FILEDIALOG_ALWAYS_SHOW_TYPES, 1)
+         wxSystemOptions::SetOption('osx.openfiledialog.always-show-types', 1)
     @endcode
     But in contrast to Windows and Unix, where the file type choice filters only
     the selected files, on Mac macOS even in this case the dialog shows all files
     matching all file types. The files which does not match the currently selected
-    file type are greyed out and are not selectable.
+    file type are greyed out and are not selectable. Also, currently the Mac macOS
+    implementation is bugged, causing all filters after the first to also have the
+    first filter applied, so ensure your first filter is a superset of all other
+    wildcard filters to avoid any issues.
```

Comment:
Sorry, "is bugged" is just not the right way to say it. If I understand the issue correctly, I would rather say something like

```
@note Due to a macOS limitation, the first filter should be a catch-all one,
      i.e. use `*` as the wildcard, as it is always applied in addition to
      all the other filters, when another one is selected.
```

Of course, if it's possible to fix this, it would be even better. @csomor Do you know if this is really a macOS limitation or just wxOSX bug?